collect_app/src/main/java/org/odk/collect/android/activities/FormMapActivity.java

```diff
@@ -207,20 +248,26 @@ private void updateInstanceGeometry() {
     private void updateMapFeatures() {
         points.clear();
         map.clearFeatures();
+        instancesByFeatureId.clear();
 
         List<MappableFormInstance> instances = viewModel.getMappableFormInstances();
         for (MappableFormInstance instance : instances) {
             MapPoint point = new MapPoint(instance.getLatitude(), instance.getLongitude());
             int featureId = map.addMarker(point, false);
 
-            int drawableId = getDrawableIdForStatus(instance.getStatus());
-            map.setMarkerIcon(featureId, drawableId);
+            updateSubmissionMarker(featureId, instance.getStatus(), featureId == viewModel.getSelectedSubmissionId());
 
             instancesByFeatureId.put(featureId, instance);
             points.add(point);
         }
     }
 
+    private void updateSubmissionMarker(int featureId, String status, boolean enlarged) {
+        int drawableId = getDrawableIdForStatus(status, enlarged);
+        map.setMarkerIcon(featureId, drawableId);
+        map.setMarkerAnchorToBottomCenter(featureId);
```

Comment:
Setting the anchor should only need to happen on marker creation. In general, the API created here feels awkward to me though I know it's tricky because of the different ways the three mapping engines represent anchors. Instead of introducing `setMarkerAnchorToBottomCenter`, how about adding an `addMarker` variant? Mapbox's `Property.ICON_ANCHOR` is a good example of introducing a type for this (see https://guides.codepath.com/android/Replacing-Enums-with-Enumerated-Annotations for more about the pattern used there). I'm thinking `addMarker(MapPoint point, boolean draggable, @ICON_ANCHOR String iconAnchor)`. The `ICON_ANCHOR` annotation can be introduced in the `MapFragment` interface with just a single value for `BOTTOM` for now. Then each `MapFragment` can test for `BOTTOM` and do whatever they need to do to pass that on.

I like that approach better because I don't think we'd ever want the anchor changing after a marker has been added so it better captures the semantics. Most importantly, it avoids the exact duplicate `setMarkerAnchorToBottomCenter` implementations and the `instanceOf` calls. I also like that it's a bit more flexible in case we need different anchors in the future (unlikely).

Reply:
I agree your approach looks better. Thanks!
Done.